spec/models/piece_spec.rb

```diff
@@ -1,5 +1,40 @@
 require 'rails_helper'
 
 RSpec.describe Piece, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  describe 'obstructed method' do
+    it 'should return nil if invalid move' do
+      piece = Piece.create(x_position: 4, y_position: 4)
+      expect(piece.obstructed?(1, 3)).to eq nil
+    end
+
+    it 'should return position if obstructed on right' do
+      piece = Piece.create(x_position: 0, y_position: 2)
```

Comment:
Then in your code, you can do this

```
game.pieces.exists?(x_position: something, y_position: something) # returns true/false
```

http://api.rubyonrails.org/v4.2.7.1/classes/ActiveRecord/FinderMethods.html#method-i-exists-3F